lib/toast.ts

```diff
@@ -0,0 +1,125 @@
+import Toastify from 'toastify-js'
+
+declare interface Toast {
+	showToast(): void;
+	hideToast(): void;
+}
+
+class ToastType {
+	static readonly ERROR = 'toast-error';
+	static readonly WARNING = 'toast-warning';
+	static readonly INFO = 'toast-info';
+	static readonly SUCCESS = 'toast-success';
+	static readonly PERMANENT = 'toast-error';
+}
+
+export interface ToastOptions {
+	/**
+	 * Defines the timeout after which the toast is closed. Set to 0 to have a persistent toast.
+	 */
+	timeout: number;
+	/**
+	 * Set to true to allow HTML content inside of the toast text
+	 * @default false
+	 */
+	isHTML: Boolean;
+	/**
+	 * Set a type of {ToastType} to style the modal
+	 */
+	type: ToastType|undefined;
+	/**
+	 * Provide a function that is called after the toast is shown
+	 */
+	callback: Function,
+	/**
+	 * Provide a function that is called when the toast is clicked
+	 */
+	onClick: Function,
+	/**
+	 * Make the toast closable
+	 */
+	close: Boolean;
+}
+
+/**
+ * Show a toast message
+ *
+ * @param text Message to be shown in the toast, any HTML is removed by default
+ * @param options
+ */
+export function showMessage(text: string, options: ToastOptions): Toast {
```

Comment:
```suggestion
export function showMessage(text: string, options?: ToastOptions): Toast {
```

Unless they are mandatory